### ISSUE TITLE
Refactor isogeny_primes.py

### ISSUE THREAD

**Issue by BarinderBanwait** [isogeny-primes/isogeny-primes#12]
The top-level function `isogeny_primes.py` is now almost 800 lines long - really the sections on TypeOnePrimes, TypeTwoPrimes, and PreTypeOneTwoPrimes should be in separate files, and their top-level functions should be imported into `isogeny_primes`.

**Comment by koffie** [comment]
Agreed. And additionally all the formal immersion stuff deserves it's own file as well I guess.

**Comment by koffie** [comment]
I think it might make sense at some time to add some unit testing, and automated code quality checking/fixing tool to this repository like black https://github.com/psf/black. Regarding the refactor, I was thinking it might make sense to start making an "isogeny_prime_filter" class. This class should store some common attributes that are used by all the different strategies/filters, and this class should be responsible to store for each epsilon type what the "remaining" primes are. So then we have one file that contains this class and has all the actual different filtering strategies. And then we have another file where we actually have the command line tool frontend depending on the input parameters. And depending what options are passed, you just call the appropriate filters.

This class based approach makes it maybe more difficult to spread things over different files. But it does allow for an easier sharing of the current state between different steps. Of course not every member function of the class needs to have the entire implementation in the file with the class itself. If certain steps are really long, then we might still put all the detailed logic in a normal funtion in another file and just make the class responsible for calling this function with the correct argument, and updating the internal state accordingly

**Comment by BarinderBanwait** [comment]
@koffie I've just pushed a major refactor directly to main; most of the sage code has been moved into a new folder `sage_code`. The main code is functional; only the `latex_helper.py` remains to be fixed, which i'll do tomorrow (probably not a major blocker for now). Do ensure that you pull this commit before doing any major code changes to prevent merge conflicts!

I also added an autopep8 feature using black in my vscode, so that might explain various other automated code refactoring.
